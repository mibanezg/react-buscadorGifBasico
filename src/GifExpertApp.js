import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({defaultCategories=[]}) => {
    //const categories = ['One Punch','Samurai X','Dragon Ball']
    const [categories, setCategories] = useState(defaultCategories);
    /*
    const handleAdd=()=>{
        setCategories([...categories,'HunterXhunter']);
        
    }
    */
    return (
        <>
            <h2>Buscador Simple</h2>
            <hr/>
            <AddCategory setCategories={setCategories}/>
            <ol>
                {
                    categories.map(category=>
                        <GifGrid 
                            key={category}
                            category={category}
                        />
                    )
                }
            </ol>
        </>
    )
}
