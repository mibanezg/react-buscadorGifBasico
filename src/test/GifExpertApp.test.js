import React from 'react';
import '@testing-library/jest-dom';
import { GifExpertApp } from '../GifExpertApp';
import { shallow } from 'enzyme';

describe('PRueba con GifExpertApp', () => {
    test('debe hacer match con snapshot', () => {
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar una lista de categorias', () => {
        const categories = ['iron man', 'xmen'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    
})
