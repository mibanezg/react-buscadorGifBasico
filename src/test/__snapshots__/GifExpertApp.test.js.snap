// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PRueba con GifExpertApp Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    Buscador Simple
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="iron man"
      key="iron man"
    />
    <GifGrid
      category="xmen"
      key="xmen"
    />
  </ol>
</Fragment>
`;

exports[`PRueba con GifExpertApp debe hacer match con snapshot 1`] = `
<Fragment>
  <h2>
    Buscador Simple
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;
